raiden-ts/src/messages/utils.ts

```diff
@@ -54,61 +84,40 @@ export function createMessageHash(message: EnvelopeMessage): Hash {
     case MessageType.LOCKED_TRANSFER:
     case MessageType.REFUND_TRANSFER:
       // hash of packed representation of the whole message
-      return keccak256(
-        concat([
-          encode(CMDIDs[message.type], 1),
-          encode(0, 3), // pad(3)
-          encode(message.nonce, 8),
-          encode(message.chain_id, 32),
-          encode(message.message_identifier, 8),
-          encode(message.payment_identifier, 8),
-          encode(message.lock.expiration, 32),
-          encode(message.token_network_address, 20),
-          encode(message.token, 20),
-          encode(message.channel_identifier, 32),
-          encode(message.recipient, 20),
-          encode(message.target, 20),
-          encode(message.initiator, 20),
-          encode(message.locksroot, 32),
-          encode(message.lock.secrethash, 32),
-          encode(message.transferred_amount, 32),
-          encode(message.locked_amount, 32),
-          encode(message.lock.amount, 32),
-          encode(message.fee, 32),
-        ]),
-      ) as Hash;
+      const packed = concat([
+        encode(CMDIDs[message.type], 1),
+        encode(message.message_identifier, 8),
+        encode(message.payment_identifier, 8),
+        encode(message.lock.expiration, 32),
+        encode(message.token, 20),
+        encode(message.recipient, 20),
+        encode(message.target, 20),
+        encode(message.initiator, 20),
+        encode(message.lock.secrethash, 32),
+        encode(message.lock.amount, 32),
+        encode(message.fee, 32),
+      ]);
+      const hashable =
```

Comment:
Nitpick: wouldn't it be clearer to `let packed`, and then 
```
if (message.type === MessageType.LOCKED_TRANSFER)
  packed = concat([packed, createMetadataHash(message.metadata)])`;
return keccak256(packed) as Hash;
```
Like, to tell: return for both is keccak of packed, but for the locked transfer, the metadata hash is appended.
This is just a suggestion in order to prevent the need for a new var (`hashable`) which has the same meaning as `packed` before hashing (the packed data which will be hashed), so you can ignore if you think it's better the way it is